datev_export_xml/views/templates.xml

```diff
@@ -0,0 +1,180 @@
+            t-att-account_no="line.account_id.code or ''"
+            t-att-cost_category_id="line.analytic_account_id.code or ''"
+            t-att-booking_text="(line.name or line.product_id.name or '')[:60]"
+            t-att-bu_code="line.tax_ids.l10n_de_datev_code or '0'"
+        />
+    </template>
+
+    <template id="export_invoice_line">
+        <t t-set="prices" t-value="line.datev_price_information()" />
+
+        <invoice_item_list
+            t-if="line.product_id"
+            t-call="datev_export_xml.export_invoice_line_item"
+            t-att-product_id="line.product_id.default_code"
+            t-att-description_short="(line.name or line.product_id.name)[:40]"
+            t-att-net_product_price="doc.datev_format_total(prices['total_excluded'], 3)"
+            t-att-order_unit="line.product_uom_id.name"
+            t-att-quantity="'%.02f' % line.quantity"
+        />
+        <invoice_item_list
+            t-elif="not line.display_type"
+            t-call="datev_export_xml.export_invoice_line_item"
+            t-att-description_short="line.name or ''"
+            t-att-net_product_price="doc.datev_format_total(prices['total_excluded'], 3)"
+            t-att-quantity="'%.02f' % line.quantity"
+        />
+    </template>
+
+    <template id="export_invoice">
+        <invoice
+            xml_data="Kopie nur zur Verbuchung berechtigt nicht zum Vorsteuerabzug"
+            xmlns="http://xml.datev.de/bedi/tps/invoice/v050"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://xml.datev.de/bedi/tps/invoice/v050 Belegverwaltung_online_invoice_v050.xsd"
+            version="5.0"
+            description="DATEV Import invoices"
+            generating_system="Odoo"
+            t-att-generator_info="doc.company_id.name"
+        >
+            <invoice_info
+                t-if="doc.datev_order_id()"
+                t-att-delivery_date="doc.datev_delivery_date()"
+                t-att-invoice_date="doc.invoice_date"
+                t-att-invoice_id="doc.datev_invoice_id()"
+                t-att-invoice_type="doc.datev_invoice_type()"
+                t-att-order_id="doc.datev_order_id()"
+            />
+            <invoice_info
+                t-else=""
+                t-att-delivery_date="doc.datev_delivery_date()"
+                t-att-invoice_date="doc.invoice_date"
+                t-att-invoice_id="doc.datev_invoice_id()"
+                t-att-invoice_type="doc.datev_invoice_type()"
+            />
+
+            <invoice_party t-call="datev_export_xml.export_party">
+                <t
+                    t-set="partner"
+                    t-value="doc.partner_id"
+                    t-if="doc.move_type in ['out_invoice', 'out_refund']"
+                />
+                <t t-set="partner" t-value="doc.company_id.partner_id" t-else="" />
+                <t t-set="account" t-value="partner.property_account_receivable_id" />
```

Comment:
This should follow similar t-if-else pattern as in supplier_party. Currently it's always setting doc.company_id.partner_id as the partner.

Reply:
@jsierp Did you test this end-to-end with a proper example ? If i remember correctly we needed to have this different, as otherwise we didn't get the correct partners in supplier bills / supplier refunds and customer invoices / customer refunds (f.e. issuing party).

@fkantelberg @marylla Can you confirm this ?

Reply:
Unfortunately I don't have a possibility to test it end-to-end atm, but setting the account in line 96 to doc.company_id.partner_id.property_account_receivable_id just seems wrong.
Consider a vendor bill. In line 95 the partner is set to doc.company_id.partner_id. Shouldn't the receivable account be always taken from the invoice partner (and not the company, because of line 95), i.e. doc.partner_id.property_account_receivable_id?
It was at least like this in previous versions of this module and I don't see why something would change in Odoo 15.

Reply:
I finally got some time to reply here and it's fine because there are `t-if` and `t-else` available. You don't have to introduce extra `<t>` tags.

The current code is equal to this if you insert the extra `<t>` tags.
```xml
<invoice_party t-call="datev_export_xml.export_party">
    <t t-if="doc.move_type in ['out_invoice', 'out_refund']">
        <t t-set="partner" t-value="doc.partner_id" />
    </t>
    <t t-else="">
        <t t-set="partner" t-value="doc.company_id.partner_id" />
    </t>
    <t t-set="account" t-value="partner.property_account_receivable_id" />
</invoice_party>
```

The difference between `invoice_party` and `supplier_party` is that the account is always set for `invoice_party` and only set for incoming invoices for `supplier_party`.

Reply:
Yes, but in version 14.0 the account for `invoice_party` was set only for outgoing invoices. I'm wondering why this was changed in this version, or am I still missing something?

Reply:
We noticed it in 15.0 and I backported it recently.

Reply:
In which commit? Here: https://github.com/OCA/l10n-germany/blob/14.0/datev_export_xml/views/templates.xml#L89C1-L101C1 the account gets set only for outgoing invoices:
```xml
<invoice_party t-call="datev_export_xml.export_party">
    <t t-if="doc.move_type in ['out_invoice', 'out_refund']">
        <t t-set="partner" t-value="doc.partner_id" />
        <t
            t-set="account"
            t-value="partner.property_account_receivable_id"
        />
    </t>
    <t t-else="">
        <t t-set="partner" t-value="doc.company_id.partner_id" />
    </t>
</invoice_party>

```

Reply:
Thanks for noticing me. Seems like I failed to cherry pick it. Either way 15.0 is the most tested version there is and should be the template for the migration not 14.0.